src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -95,6 +95,15 @@ class LinalgBackendEigen : public LinalgBackendBase
 	DEFINE_FOR_ALL_PTYPE(BACKEND_GENERIC_IN_PLACE_ADD, SGMatrix)
 	#undef BACKEND_GENERIC_IN_PLACE_ADD
 
+	/** Implementation of @see LinalgBackendBase::apply */
+	#define BACKEND_GENERIC_IN_PLACE_APPLY(Type, Container) \
+	virtual void apply(SGMatrix<Type>& a, Container<Type>& b, Container<Type>& result) const \
```

Comment:
@karlnapf `matmul` is matrix * matrix but I didn't see why we can't merge `matrix * vector` and `matrix * matrix` into one method. We can throw an error if `type(B) = SGVector && transpose_B = true"